xdem/spstats.py

```diff
@@ -25,6 +27,109 @@
     import skgstat as skg
     from skgstat import models
 
+def interp_nd_binning(df: pd.DataFrame, list_var_names: Union[str,list[str]], statistic : Union[str, Callable[[np.ndarray],float]] = nmad,
+                      min_count: Optional[int] = 100) -> Callable[[tuple[np.ndarray, ...]], np.ndarray]:
+    """
+    Estimate an interpolant function for an N-dimensional binning. Preferably based on the output of nd_binning.
+    For more details on the input dataframe, and associated list of variable name and statistic, see nd_binning.
+
+    If the variable pd.DataSeries corresponds to an interval (as the output of nd_binning), uses the middle of the interval.
+    Otherwise, uses the variable as such.
+
+    Workflow of the function:
+    Fills the no-data present on the regular N-D binning grid with nearest neighbour from scipy.griddata, then provides an
+    interpolant function that linearly interpolates/extrapolates using scipy.RegularGridInterpolator.
+
+    :param df: dataframe with statistic of binned values according to explanatory variables (preferably output of nd_binning)
+    :param list_var_names: explanatory variable data series to select from the dataframe (containing interval or float dtype)
+    :param statistic: statistic to interpolate, stored as a data series in the dataframe
+    :param min_count: minimum number of samples to be used as a valid statistic (replaced by nodata)
+    :return: N-dimensional interpolant function
+
+    :examples
+    Using a dataframe created from scratch
+    >>> df = pd.DataFrame({"var1": [1, 1, 1, 2, 2, 2, 3, 3, 3], "var2": [1, 2, 3, 1, 2, 3, 1, 2, 3], "statistic": [1, 2, 3, 4, 5, 6, 7, 8, 9]})
+        In 2 dimensions, the statistic array looks like this
+        array([
+            [1, 2, 3],
+            [4, 5, 6],
+            [7, 8, 9]
+            ])
+    >>> fun = interp_nd_binning(df, list_var_names=["var1", "var2"], statistic="statistic", min_count=None)
+    Right on point.
+    >>> fun((2, 2))
+        array(5.)
+    Interpolated linearly inside the 2D frame.
```

Comment:
These can be expressed as comments after the function call, e.g. `>>> fun((1.5, 1.5))  # Interpolated linearly inside the 2D frame.`

Reply:
Done in 84af672